### PR TITLE
minor fix + enhanced test for skipped extensions

### PR DESCRIPTION
**Pull request by boegel**
@Flamefire https://github.com/easybuilders/easybuild-framework/pull/3254

### CODE REVIEW

File: easybuild/framework/easyblock.py [hunk 1]
```diff
@@ -2156,7 +2156,7 @@ def extensions_step(self, fetch=False):
             # always go back to original work dir to avoid running stuff from a dir that no longer exists
             change_dir(self.orig_workdir)
 
-            tup = (ext.name, ext.version, idx+1, exts_cnt)
+            tup = (ext.name, ext.version or '', idx+1, exts_cnt)
```
Comment (Flamefire) [inline easybuild/framework/easyblock.py:2159]: Can there be a version of `None` at this point? How? Other uses of `.version`: `topdir = os.path.join(self.builddir, '%s-%s' % (self.name, self.version))` so this looks like omitting the `or...` is save

**Comment by Flamefire** [comment]
Thanks for that!

To reproduce the issue we need an extension *after*  the skipped ones. Only that shows that those become out of sync. Could you add one at https://github.com/Flamefire/easybuild-framework/pull/1/files#diff-338b803c935bc2416223a2c8baf4c500R836 or want me to do this after merging?

File: easybuild/framework/easyblock.py (hunk 1 above)
Comment (boegel) [reply easybuild/framework/easyblock.py:2159]: We have situations where extensions are specified only by name, which just serves as a check that these extensions are installed (and then the version is indeed `None`).

Try removing the `or ''`, and run the test with `python -O -m test.framework.easyblock skip_ext`

**Comment by boegel** [comment]
> Thanks for that!
> 
> To reproduce the issue we need an extension _after_ the skipped ones. Only that shows that those become out of sync. Could you add one at https://github.com/Flamefire/easybuild-framework/pull/1/files#diff-338b803c935bc2416223a2c8baf4c500R836 or want me to do this after merging?

Without the fix the counting shows `(1/3)` while only one extension is actually being installed (two are skipped).

I can try adding another (versionless) extension, but actually installing an extension (one with a version) is difficult here.

File: easybuild/framework/easyblock.py (hunk 1 above)
Comment (Flamefire) [reply easybuild/framework/easyblock.py:2159]: Ah, ok. R seems to have some of those.

**Comment by Flamefire** [comment]
Thanks!

**Pull request merged**